core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -122,7 +126,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
           context.reply(RegisteredExecutor)
-
+          latestTokens.foreach(x => context.reply(NewTokens(x)))
```

Comment:
Yeah, but the executor won't do anything if it doesn't get a `RegisteredExecutor` reply. Your style was making things worse by adding this extra case where `RegisteredExecutor` could arrive but the next message could be lost.

Anyway, the issue is mostly theoretical.